### PR TITLE
Origin/trading functionality

### PR DESCRIPTION
**Pull request by JustasAn**
User can select his own tradable items on ad of type "Keitimas" and write a message for some additional information. Owner can then open detailed ad page and select his favorite offer for trading.
After he selects an offer, an email is sent to the user with a url, where he can enter his details so that the owner could contact him.

Testing:

Create a single or more ads with type "Keitimas"
Go to another ad that has a type "Keitimas"
Sumbit your offer
In the owner page of that item, your offer should appear
As owner submit that you are accepting that offer
Trader receives email
Check email and check if owner ad status in DB is 2 and trader ad status is in 3
Enter details after clicking url in email
Check email again and check if ad status in DB is 3
Check if ad does not appear in home page

### CODE REVIEW

File: ClientApp/src/components/ItemViewPage/ItemViewPage.js [hunk 1]
```diff
@@ -50,6 +50,7 @@ export const ItemViewPage = () => {
                 try {
                     const response = await axios.get('api/item/getUserItems');
                     setUserItems(response.data);
+                    console.log(userItems);
```
Comment (Ostinas) [inline ClientApp/src/components/ItemViewPage/ItemViewPage.js]: istrink sita

File: Repositories/Item/ItemRepo.cs [hunk 2]
```diff
@@ -646,5 +646,74 @@ public async Task<List<UserViewModel>> GetLotteryParticipants(int itemId)
...
+                    {
+                        Id = id,
+                        Message = message,
+                        Name = name,
+                        Description = description,
+                        Location = location,
+                        EndDateTime = end_time,
+                        Images = await _imageRepo.GetByAd(Convert.ToInt32(reader["fk_offer_item"])),
+                        User = user
+                    };
```
Comment (Ostinas) [inline Repositories/Item/ItemRepo.cs]: cia gali sumazinti ta visa bloka tiesiogiai nuskaitydamas atributus. Pvz: 
Id = reader.GetInt32("fk_offer_item"),
Message = reader.GetString("offer_message"),
ir tt

File: Repositories/Item/ItemRepo.cs [hunk 3]
```diff
@@ -1,4 +1,5 @@
 ﻿using MySql.Data.MySqlClient;
+using MySqlX.XDevAPI.Common;
```
Comment (Ostinas) [inline Repositories/Item/ItemRepo.cs]: labai random using'as atrodo, reikia jo?

**Review by Ostinas — CHANGES_REQUESTED** [review]

**Review by Ostinas — APPROVED** [review]

**Review by Rokki-R — APPROVED** [review]
All gucci

**Pull request merged**